### PR TITLE
Adding old sidebar extension

### PR DESCRIPTION
**Pull request by estufar**
I've finally gotten around to to adding my sidebar fix! I've renamed it to old_sidebar to be more consistent with other extensions' names.

### CODE REVIEW

**Comment by nightpool** [comment]
sick! super exited to get this onboard.

we've had some people who were having problems with editable reblogs and
your extension at the same time, so whoever reviews this pr should make
sure to check out #100 and follow the reproduction steps there.

also we'll probably have to brainstorm some new logic that will replace
people who have installed the thirdparty extension with this one, and make
sure no-one installs this one as well as the old one. this isn't really
your problem, but it needs to be done before we can pull this. (ideally the
same logic would suffice for this usecase, the usecase where we rename an
extension, and the usecase where we remove an extension)

I'll make a new ticket to track this blocker.

**Comment by Wolvan** [comment]
@nightpool Couldn't we just do the same thing Xenix did back in the day with XIM? Didn't he somehow automatically make XKit delete XIM?
Also important for #410

**Comment by nightpool** [comment]
@Wolvan did he?

I just made #410 now to track this issue—did you mean to reference something else?

**Comment by Wolvan** [comment]
No, I referenced #410 so my message does show there as well. But yeah, he somehow did a force delete of the XIM extension back when he took it down

**Comment by nightpool** [comment]
@wolvan its in xkit_preferences line 264. we need logic that would replace
this extension with a different one though, not just removing it.

and ideally instead of it being hardcoded, xkit would pull this list from
the dist json.

File: Extensions/old_sidebar.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			value: false
+		}
+	},
+	
+	run: function() {
+		XKit.tools.init_css("old_sidebar");
+		this.running = true;
+		
+		var movesidebar = true;
+		if (XKit.extensions.old_sidebar.preferences.dashonly.value === true) {
```
Comment (Wolvan) [inline Extensions/old_sidebar.js]: I'd suggesst you remove the "=== true" check here

File: Extensions/old_sidebar.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			default: false,
+			value: false
+		}
+	},
+	
+	run: function() {
+		XKit.tools.init_css("old_sidebar");
+		this.running = true;
+		
+		var movesidebar = true;
```
Comment (Wolvan) [inline Extensions/old_sidebar.js]: Instead of having a variable movesidebar just do `` return; `` where you otherwise use `` movesidebar = false; ``

**Comment by Wolvan** [comment]
@estufar Is there a reason why you use vanilla JS and not JQuery?

**Comment by estufar** [comment]
@Wolvan It's mainly just that I didn't realize JQuery was there when I originally wrote it a few months ago and I just haven't updated it to use it. I could go ahead and update it to use JQuery if we want it to be more consistent with the other extensions.

**Comment by nightpool** [comment]
I think we're fine without the jquery.

LGTM as soon as we have the infrastructure in place to replace the old
extension with this one

**Comment by dlmarquis** [comment]
We could do things the janky way and push an update to delete/comment out the meaty parts of Old Stats, although from experience I can say that not much happens when you've got both loaded (unless something major changed when you fixed the dashboard click issue).

This is tangential, but @estufar, are you okay with us adding features (such as styling or an option to bring the menu back), or would you want to keep control over this extension?

**Comment by estufar** [comment]
I'm all for people adding features to the extension

**Comment by Wolvan** [comment]
@nightpool @estufar I know that you renamed the extension to be more consistent with everything else, but I think we can stick with sidebar_fix since it technically is an update to an already existing extension, if you know what I mean

**Comment by estufar** [comment]
@Wolvan I'd just mainly renamed the extension to remove my username from being part of it (it was estufars_sidebar_fix which definitely wasn't consistent). Other than that, I'm good with calling it old_sidebar or sidebar_fix, whichever would work better.

**Comment by Wolvan** [comment]
Just saying since we could have deployed it automatically that way. What about you making an "update" to the ESF extension that doesn't have any functionality anymore and just tells the user to go to the extension gallery to download the new SF?

**Comment by estufar** [comment]
Yeah, I don't really know the best way to go about it. I could definitely just rename this back to ESF and have the updating process work better, but I think it would look a little weird having that one extension with my name in the file. But then again, nobody but us is really going to see that so I should probably just go ahead and change it back and not worry about it.

**Comment by Wolvan** [comment]
@estufar sounds good to me, but I would like to hear @nightpool's opinion on this as well

**Comment by dlmarquis** [comment]
@Wolvan we should keep the old file name, but leave the title as Old Sidebar so that the extension gets overwritten, but the UI name isn't weird. Like how RDO is better_reblogs.js and Header Options is classic_header.js

**Comment by Wolvan** [comment]
@dlmarquis that is what I've been saying

**Comment by dlmarquis** [comment]
Ah. I wasn't sure if you wanted both to change or not.

**Comment by nightpool** [comment]
this is ready for merge as soon as the file is renamed to estufars_sidebar_fix, and code is added to fix #100.

the code would be 
```javascript
if(document.location.href.indexOf('://www.tumblr.com/reblog') != -1) {
         return;
}
```
right before line 37

**Comment by Wolvan** [comment]
Since @nightpool's changes have been added, I give this a lgtm
@homu r+

**Comment by nightpool** [comment]
Forgot to ask this earlier, but this had a version bump from the released version, right?

**Comment by estufar** [comment]
I somehow ended up changing it from 1.1 to 1.1.0, but I'm not sure whether that counts as a version bump. I'm also not quite sure how I ended up changing the version number to that.

**Comment by Wolvan** [comment]
@estufar 1.1 is parsed as 1.1.0. So no, it's not a version bump. BUT people who use your extension should already have 1.1, so I don't see the harm until next update. And all other people download the 1.1.0 version

**Comment by nightpool** [comment]
@wolvan this version fixes #100 though.

**Comment by nightpool** [comment]
I'll manually bump it when I get back to my computer.

**Comment by Wolvan** [comment]
@nightpool Oh I see? Should I do the version bump real quick?

**Comment by nightpool** [comment]
Yeah sure. I'm about to go to bed anyway

**Pull request merged**